code/modules/reagents/newchem/newchem_procs.dm

```diff
@@ -51,14 +51,15 @@
 				continue
 		//If you got this far, that means we can process whatever reagent this iteration is for. Handle things normally from here.
 		if(M && R)
+			R.on_mob_life(M)
```

Comment:
The reason this is moved here is so that some of the overdosing of reagents has a high impact.

For example, if you get stunned by overdosing on Meth, fall on the floor, then the reagent processes, the meth is going to reduce the stun duration; when it's at this location, it makes a number of the overdose effects *much* more punishing.